pkg/odo/cli/registry/registry.go

```diff
@@ -0,0 +1,213 @@
+	if err != nil {
+		return err
+	}
+
+	if o.DevfileList.DevfileRegistries == nil {
+		log.Warning("Please run 'odo preference registry add <registry name> <registry URL>' to add registry for listing devfile components\n")
+	}
+
+	return nil
+}
+
+// Validate validates the ListOptions based on completed values
+func (o *ListOptions) Validate() error {
+	if len(o.DevfileList.Items) == 0 {
+		return fmt.Errorf("no deployable components found")
+	}
+	return nil
+}
+
+// Run contains the logic for the command associated with ListOptions
+func (o *ListOptions) Run(ctx context.Context) (err error) {
+	o.printDevfileList(o.DevfileList.Items)
+	return nil
+}
+
+func NewCmdRegistry(name, fullName string) *cobra.Command {
+	o := NewListOptions()
+
+	var listCmd = &cobra.Command{
+		Use:     name,
+		Short:   "List all components from the Devfile registry",
+		Long:    "List all components from the Devfile registry",
+		Example: fmt.Sprintf(Example, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			genericclioptions.GenericRun(o, cmd, args)
+		},
+	}
+
+	clientset.Add(listCmd, clientset.REGISTRY)
+
+	// Flags
+	listCmd.Flags().StringVar(&o.filterFlag, "filter", "", "Filter based on the name of the component")
+	listCmd.Flags().StringVar(&o.devfileFlag, "devfile", "", "Only show components from the specific Devfile stack")
+	listCmd.Flags().BoolVar(&o.detailsFlag, "details", false, "Show details of each component")
+
+	// Add a defined annotation in order to appear in the help menu
+	listCmd.Annotations["command"] = "main"
+	listCmd.SetUsageTemplate(odoutil.CmdUsageTemplate)
+
+	return listCmd
+}
+
+func (o *ListOptions) printDevfileList(DevfileList []registry.DevfileStack) {
+
+	// Create the table and use our own style
+	t := table.NewWriter()
+	rowConfigAutoMerge := table.RowConfig{AutoMerge: true}
+
+	// If passed int he details flag, we'll be using a different style of table output.
+	if o.detailsFlag {
+		t.SetStyle(table.StyleLight)
+		t.Style().Options.SeparateRows = true
+		t.Style().Color.Header = text.Colors{text.Bold}
+		t.Style().Box.PaddingLeft = " "
+		t.Style().Box.PaddingRight = " "
+	} else {
+		t.SetStyle(table.Style{
+			Box: table.BoxStyle{
+				PaddingLeft:  " ",
+				PaddingRight: " ",
+			},
+			Color: table.ColorOptions{
+				Header: text.Colors{text.FgHiGreen, text.Underline},
+			},
+			Format: table.FormatOptions{
+				Footer: text.FormatUpper,
+				Header: text.FormatUpper,
+				Row:    text.FormatDefault,
+			},
+			Options: table.Options{
+				DrawBorder:      false,
+				SeparateColumns: false,
+				SeparateFooter:  false,
+				SeparateHeader:  false,
+				SeparateRows:    false,
+			},
+		})
+	}
+	t.SetOutputMirror(log.GetStdout())
+
+	// If we pass in the details flag, we will manipulate the table as well as create some
+	if o.detailsFlag {
+		t.AppendHeader(table.Row{"NAME / VERSION", "REGISTRY / URL", "LANGUAGE / TAGS", "DEV", "DEPLOY", "DEBUG", "DESCRIPTION", "STARTER PROJ."}, rowConfigAutoMerge)
```

Comment:
IMO, keeping the `REGISTRY/URL` column a bit later might be better. Saying this because in a real registry, there are going to be a bunch of stacks and the registry info might not even be immediately useful to users not working with multiple registries. 

In the screenshot, you can see that I have only one registry and that is occupying a good amount of space which causes info about starter projects difficult to get.
![Screenshot from 2022-04-18 15-45-47](https://user-images.githubusercontent.com/713521/163795161-0e94d063-1fb9-4cbf-8331-97eae399d174.png)